app.py

```diff
@@ -45,6 +47,30 @@ def init_app(self):
                                      self.create_panel, 
                                      {"type": "panel", 
                                       "short_name": "shotgun_panel"})
+
+    @property
+    def emitter(self):
+        """
+        The Emitter object utilized by the app. Any QSignals emitted
+        by the app will come from this object.
+        """
+        return self._emitter
+
+    @property
+    def context_change_allowed(self):
+        """
+        Specifies that context changes are allowed.
+        """
+        return True
+
+    def post_context_change(self, old_context, new_context):
+        """
+        Runs after a successful change of context.
+
+        :param old_context: The context prior to the context change.
+        :param new_context: The new context after the context change.
+        """
+        self.emitter.context_changed.emit(old_context, new_context)
```

Comment:
I am wondering about the behaviour here. You can pop up a shotgun panel via the `panel_app_object.navigate()` method - the intention here is that you can pop up a window with info about stuff - say for an object in the breakdown or scene or something. Imagine a maya scene with multiple panel dialog windows open, showing notes for different assets, or a future timeline based thing (hiero/flame) where you see notes for different shots. At the point of the context switch you don't want to flip all those guys around to show the new context. You only want to flip the main "panel" of which there can only be one active instance.

I was imagining the scope of the ticket here was different - that we wouldn't even navigate the "main" panel automatically - i thought all we would do was to simply change the location where the home button would take you, *should you press the button*. Rule of least surprise I guess. What if you are in the process of writing a note or something in the panel as you switch context (by opening a new file in maya say?). We should clarify with @robblau. Should the panel navigate automatically on context switch or should we simply make sure that if a user presses the home button, you always navigate to the current context? Personally, as a user, I think i would like as little auto navigation as possible. It's easy enough to press the home button if i need to.

Assuming the former, I guess the implementation should only affect the one panel, not any dialogs. We should probably just keep track of the panel widget we create (similar to what we do with `self._current_dialog`) and then maybe we can just call a method directly on that object?

```python
# tell any active shotgun panel to refocus itself
if self._current_panel:
   self._current_panel.navigate_home() # _on_home_clicked() renamed
```